### DISCUSSION TITLE
How to use data option in dialogs

### DISCUSSION THREAD

**Discussion by loxy** [taiga-family/taiga-ui#782] [Q&A] [answered]
Hi!

I want to use the `DialogService` to show a custom dialog from template. But I need some custom data inside of it. I saw there is a data attribute in `TuiDialogOptions`, but I didn't get access to it in the template. Something like:

`<ng-template #template let-observer let-data>`

Do I have to create a custom dialog in order to use it?

Best regards,
Kersten

**Comment by waterplea** [comment, marked as answer]
Template context in Angular works like this
1. You can access `$implicit` key by just writing `let-whatever` 
2. You can access all other keys by explicitly naming them:
```html
<ng-template #template let-nameToUse="data">
  {{nameToUse}}
</ng-template>
```

**Reply by loxy** [reply]
Right. Silly question.

**Comment by HuzaifaJH** [comment]
```
    this.dialogs
      .open(
        new PolymorpheusComponent(RfiDetailsDialogComponent),
        {
          data: {
            rfi: rfi,
          },
          size: 'auto',
          dismissible: false,
        },
      )
      .subscribe();
```

I am passing this rfi to my rfiDetailsDialogComponent, but I cannot find it in the dialog component in context.
Can anyone help me with this?
